### DISCUSSION TITLE
Signing out user programmatically or via dashboard

### DISCUSSION THREAD

**Discussion by liaujianjie** [supabase/supabase#7267] [Questions] [answered]
Is there currently a way to sign a user out programmatically or via the Supabase dashboard?

**Comment by GaryAustin1** [comment, marked as answer]
There is not built in way.  signOut client call basically deletes all the user's refresh tokens in the auth.refresh_token table.  It also deletes any information in the browser it is run on.  Any other devices will keep working until the jwt expires.

So you could probably write a postgres function to take a user id and delete the refresh tokens.  This would prevent that user from refreshing any jwts they have when they expire.   They then have to signIn again to get a new refresh token.
